src/sdk.ts

```diff
@@ -393,15 +395,24 @@ export default class SDK {
         mxy.client,
         mxy.user,
         mxy.messenger,
+        mxy.purchaseClient,
         mxy.analytics,
         mxy.loadPromise,
         mxy.SKUs,
-        mxy.debugOptions
+        mxy.debugOptions,
       );
 
       mxy.SDKClients[identifier] = this;
     }
 
+    if (mxy.purchaseClientEnabled) {
+      this.purchaseClient.onUserPurchase((tx: Transaction) => {
+        if (tx.initiator.toLowerCase() === "current_user") {
```

Comment:
This check doesn't need to be done here, as it is already performed inside `onUserPurchase`.